### DISCUSSION TITLE
Why err.response is typed as any instead of Response?

### DISCUSSION THREAD

**Discussion by frederikhors** [urql-graphql/urql#1460] [Q&A] [answered]
I'm using the `retryExchange` to do some work for me:

```ts
const exchanges = [
  // ... exchanges
  retryExchange(retryOptions)
];

const retryOptions = {
  // ...some options
  retryIf: (err: CombinedError) => {
    if (err.response) {
      const response = err.response as Response;
      if ([401].includes(response.status)) {
        return Logout();
      }
    }
    if (err.networkError) {
      showMessage(err.networkError.message)
      return err.networkError;
    }
  },
};
```

As you can see I need to cast `err.response` as `Response` because it's `any`.

Hence my question: **why `err.response` is `any`?**

Is there a better way to write this?

**Comment by kitten** [comment, marked as answer]
That's because we don't want to type it or guarantee a type because the `response` may be set to any arrogant that describes the original result of a request, be it for WebSockets or other requests. In other words, because the transport protocol may be swapped out. We didn't want to set this to `unknown` for now because that may make it even more confusing, but yea, basically it isn't guaranteed to be a `FetchResponse`

**Reply by frederikhors** [reply]
Ok. As always thanks!

**Reply by frederikhors** [reply]
Maybe `unknown` is better for linters warnings...
